tests/query/snql/test_invalid_queries.py

```diff
@@ -0,0 +1,32 @@
+import pytest
+from parsimonious.exceptions import IncompleteParseError
+
+from snuba.datasets.factory import get_dataset
+from snuba.query.snql import parse_snql_query
+
+test_cases = [
+    # below are all cases that are not parsed completely
+    # i.e. the entire string is not consumed
+    pytest.param(
+        "MATCH(blah)WHEREa<3COLLECT4-5,3*g(c),cBYd,2+7ORDERBYfDESC",
+        IncompleteParseError,
+        id="ORDER BY is two words",
+    ),
+    pytest.param(
+        "MATCH(blah)WHEREa<3COLLECT4-5,3*g(c),cBYd,2+7ORDER BYfDESC",
```

Comment:
mind adding the spaces between match, where and collect, if you are testing the ORDER BY, the rest of the query can still be valid.